xdsl/transforms/memref_streamify.py

```diff
@@ -29,19 +28,22 @@ def match_and_rewrite(
 
         init_indices = set(index.data for index in op.init_indices)
 
-        # Currently can only stream memrefs that are not inout
+        # Can only stream memrefs that are not inout
+        input_count = len(op.inputs)
         streamable_input_indices = tuple(
-            (index, cast(memref.MemRefType[Attribute], value_type).element_type)
-            for index, value in enumerate(op.inputs)
-            if isinstance(value_type := value.type, memref.MemRefType)
-            and op.body.block.args[index].uses
+            (index, op.body.block.args[index].type)
```

Comment:
```suggestion
            (index, arg.type)
```
(if my Python is not incorrect)